### PR TITLE
Allow force deleting agent pools

### PR DESCRIPTION
**Pull request by komalali**
Sometimes, a user may want to force delete an agent pool, even if it is associated with existing deployment settings. This is already supported in the API, bringing the functionality to PSP.

### CODE REVIEW

File: provider/pkg/provider/agent_pool_test.go
Comment (nyobe) [inline provider/pkg/provider/agent_pool_test.go:67]: isn't this just testing that the mock returns what you told it to return? 😞
I don't think this is useful

File: examples/ts-webhooks/index.ts
Comment (komalali) [inline examples/ts-webhooks/index.ts:42]: Needed an update due to new URL validation for webhooks

File: provider/pkg/provider/agent_pool_test.go
Comment (komalali) [reply provider/pkg/provider/agent_pool_test.go:67]: Yeah that's fair. Anything that's not a full integration test has been somewhat tricky to test in the provider. Let me spend some time figuring this out.

File: provider/pkg/internal/pulumiapi/agent_pools.go
Comment (dschaller) [inline provider/pkg/internal/pulumiapi/agent_pools.go:110]: super nit: should we consider making options like this a struct in case we need to add more later?

File: provider/pkg/provider/agent_pool.go
Comment (dschaller) [inline provider/pkg/provider/agent_pool.go:36]: Might want to make `forceDestroy` a constant

Comment (komalali) [reply provider/pkg/provider/agent_pool.go:36]: Hmm, not sure it makes sense to make one input a constant and not any of the others... so I'd leave that to a future optimization since this is not the pattern anywhere in the provider.

File: provider/pkg/provider/agent_pool_test.go
Comment (nyobe) [reply provider/pkg/provider/agent_pool_test.go:67]: it seems like most of the other tests in the package are the same way 🤷‍♀️ 
if you figure out something I'd love to know for #436 !

**Review by nyobe — APPROVED** [review]

File: provider/pkg/provider/agent_pool_test.go
Comment (komalali) [reply provider/pkg/provider/agent_pool_test.go:67]: Yeah I think I'm going to merge this just to unblock Scala but this is a good question that we do need an answer to beyond E2E tests

**Pull request merged**
